yargs.js

```diff
@@ -669,6 +692,42 @@ function Yargs (processArgs, cwd, parentRequire) {
     return detectLocale
   }
 
+  var hasOutput = false
+  var exitError = null
+  // maybe exit, always capture
+  // context about why we wanted to exit.
+  self.exit = function (code, err) {
+    hasOutput = true
+    exitError = err
+    if (exitProcess) process.exit(code)
+  }
+
+  // we use a custom logger that buffers output,
+  // so that we can print to non-CLIs, e.g., chat-bots.
+  var silent = false
+  var _logger = {
+    log: function () {
+      var args = Array.prototype.slice.call(arguments)
+      if (!silent) console.log.apply(console, args)
+      hasOutput = true
+      output += args.join(' ')
```

Comment:
For consistency with `console.log` and `console.error`, I think we need to include line feeds too, something like:

```js
if (output.length) output += '\n'
output += args.join(' ')
```

Same for `_logger.error`. Otherwise the spacing in `output` is not _exactly_ what would have been output to the console. What do you think?